### ISSUE TITLE
Support switch buttons

### ISSUE THREAD

**Issue by cpetrov** [eclipsesource/tabris-js#200]
Support creating switch buttons.

Example (Android switch):
![Android Switch](http://developer.android.com/images/ui/switch.png)

**Comment by jonek** [comment]
We should support this widget with Tabris.js 1.0.

**Comment by mpost** [comment]
Here is the initial version of the protocol definition:

``` javascript
tabris.registerWidget("Switch", {
  _type: "tabris.Switch",
  _events: {
    checked: true
  },
  _properties: {
    checked: {type: "boolean", nocache: true},
    textChecked: {type: "string"},
    textUnchecked: {type: "string"}
  }
});
```

**Comment by patrykmol** [comment]
iOS switch has no text in it.

**Comment by mpost** [comment]
@patrykmol In android material design there is no text either. On android 4.x the text is applied. So the native platform could make use of the text or not.

**Comment by mpost** [comment]
Btw... i have created a self hosted snippet that can act as a starting point to implement the feature:

``` javascript
var MARGIN = 16;

var page = tabris.create("Page", {
  title: "Switch",
  topLevel: true
});

tabris.registerWidget("Switch", {
  _type: "tabris.Switch",
  _events: {
    checked: true
  },
  _properties: {
    checked: {type: "boolean", nocache: true},
    textChecked: {type: "string"},
    textUnchecked: {type: "string"}
  }
});

tabris.create("Switch", {
  id: "switch",
  foreground: "red",
  layoutData: {left: MARGIN, top: MARGIN},
  checked: true,
  textChecked: "Yo!",
  textUnchecked: "Nope"
}).on("checked", function(event) {
  page.find("#checkedStateEvent").set("text", event.checked ? "State: checked" : "State: unchecked");
}).appendTo(page);

tabris.create("TextView", {
  id: "checkedStateEvent",
  text: "State: checked",
  layoutData: {left: ["#switch", MARGIN], top: MARGIN, baseline: "#switch"}
}).appendTo(page);

tabris.create("Button", {
  id: "getStateButton",
  text: "Get state",
  layoutData: {left: MARGIN, top: ["#switch", MARGIN]}
}).on("select", function() {
  page.find("#checkedStateManual").set("text", "Checked: " + page.find("#switch").get("checked"));
}).appendTo(page);

tabris.create("TextView", {
  id: "checkedStateManual",
  text: "Checked: true",
  layoutData: {left: ["#getStateButton", MARGIN], top: ["#switch", MARGIN], baseline: "#getStateButton"}
}).appendTo(page);

tabris.create("Button", {
  id: "setUncheckedButton",
  text: "Set unchecked",
  layoutData: {left: MARGIN, top: ["#getStateButton", MARGIN]}
}).on("select", function() {
  page.find("#switch").set("checked", false);
}).appendTo(page);

tabris.create("Button", {
  text: "Set checked",
  layoutData: {left: ["#setUncheckedButton", MARGIN], top: ["#getStateButton", MARGIN]}
}).on("select", function() {
  page.find("#switch").set("checked", true);
}).appendTo(page);

page.open();
```

**Comment by ralfstx** [comment]
We've discussed the API and came up with the following changes:

* It's confusing that the `checked` event is also triggered when the switch is *unchecked*. We agreed to rename it to `toggle`.
* The API should be consistent with `ToggleButton` and `CheckBox`. Thus, we should have a property `selection`. Since `checked` is more intuitive, we should support it as an alias, however, this alias should be supported on all buttons with a checked state.

**Comment by mpost** [comment]
@ralfstx We should also add the `checked` to the RadioButton i'd think.

**Comment by ralfstx** [comment]
@mpost yes, agree. I've opened #515 for an alias property.

**Comment by ralfstx** [comment]
We didn't discuss the event properties yet: the `toggle` event needs to have a boolean `checked` property. This is needed to emulate the `select` and `change:selection` event.

**Comment by ralfstx** [comment]
Seems there is a consensus to remove the `textChecked` and `textUnchecked` properties. They are only supported on Android 4 and are going to be replaced even there by an update to the Material design. @mpost correct me if I'm wrong.

**Comment by mpost** [comment]
That is correct. Go ahead with the removal.

**Comment by ralfstx** [comment]
Implemented in 7e0aac169a548a4e2ece2f8b551a939e00f6c20d.
